Fail if merge to enterprise causes compilation issues (#4027)
citusdata/citus: ci/README.md

This check exists to make sure that we can always merge the `master` branch of
`community` into the `enterprise-master` branch of the `enterprise` repo.
There are two conditions in which this check passes:

> 1. There are no merge conflicts between your PR branch and `enterprise-master` and `enterprise-master` compiles with your changes.

Reviewer: ```suggestion
1. There are no merge conflicts between your PR branch and `enterprise-master` and 
   after this merge the code compiles.
```

2. There are merge conflicts, but there is a branch with the same name in the
   enterprise repo that:

>    1. Contains the last commit of the community branch with the same name.

Reviewer: add a third item it in this list with something like:
```
3. After merging the code can be compiled
